### PR TITLE
Re-apply changes from commit ec9071116172adef6adde0a384cff2c1cce28abd

### PR DESCRIPTION
**Pull request by ClementKunz**
Reapply "Merge branch 'develop' of github.com:Infomaniak/desktop-kDrve into develop"

This reverts commit 387d82c14abe108ec78dbd2d69b92087febc6b17.

# Conflicts:
#	src/libsyncengine/update_detection/file_system_observer/localfilesystemobserverworker.cpp

### CODE REVIEW

File: src/libsyncengine/update_detection/file_system_observer/folderwatcher_mac.cpp [hunk 1]
```diff
@@ -40,17 +40,17 @@ static void callback([[maybe_unused]] ConstFSEventStreamRef streamRef, void *cli
...
+            | kFSEventStreamEventFlagItemRenamed // also coming for moves to trash in finder
+            | kFSEventStreamEventFlagItemModified // for content change
+            | kFSEventStreamEventFlagItemChangeOwner; // for rights change
 
     std::list<std::pair<std::filesystem::path, OperationType>> paths;
-    CFArrayRef eventPaths = (CFArrayRef)eventPathsVoid;
+    CFArrayRef eventPaths = (CFArrayRef) eventPathsVoid;
     for (int i = 0; i < static_cast<int>(numEvents); ++i) {
-        OperationType opType;
+        OperationType opType = OperationType::None;
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/update_detection/file_system_observer/folderwatcher_mac.cpp:53]: **Suggestion.** Make a small function that computes `opType` and write units test for it.

File: src/libcommon/utility/types.cpp [hunk 2]
```diff
@@ -51,6 +51,8 @@ std::string toString(NodeType e) {
             return "File";
         case NodeType::Directory:
             return "Directory";
+        default:
```
Comment (luc-guyot-infomaniak) [inline src/libcommon/utility/types.cpp:54]: **Suggestion.** Add a unit test that raises the default clause.

File: src/libcommon/utility/types.cpp [hunk 3]
```diff
@@ -68,6 +70,8 @@ std::string toString(OperationType e) {
             return "Delete";
         case OperationType::Rights:
             return "Rights";
+        default:
+            return "No conversion to string available";
```
Comment (luc-guyot-infomaniak) [inline src/libcommon/utility/types.cpp:74]: Idem.

File: src/libsyncengine/update_detection/file_system_observer/folderwatcher_mac.cpp (hunk 1 above)
Comment (ClementKunz) [reply src/libsyncengine/update_detection/file_system_observer/folderwatcher_mac.cpp:53]: Done in commit f5bb0a5

File: src/libcommon/utility/types.cpp (hunk 2 above)
Comment (ClementKunz) [reply src/libcommon/utility/types.cpp:54]: Done in commit [3221738](https://github.com/Infomaniak/desktop-kDrive/pull/304/commits/3221738991d4803921d7a885541deb5a3862d66e)

File: test/libcommon/utility/testtypes.cpp [hunk 4]
```diff
@@ -34,6 +36,17 @@ void TestTypes::testStreamConversion() {
...
+    CPPUNIT_ASSERT_EQUAL(std::string("Directory(2)"), toStringWithCode(static_cast<NodeType>(2)));
+    CPPUNIT_ASSERT(Utility::startsWith(toStringWithCode(static_cast<NodeType>(3)), noConversionStr));
+
+    CPPUNIT_ASSERT_EQUAL(std::string("None(0)"), toStringWithCode(OperationType::None));
+    CPPUNIT_ASSERT_EQUAL(std::string("Create(1)"), toStringWithCode(OperationType::Create));
+    CPPUNIT_ASSERT_EQUAL(std::string("Move(2)"), toStringWithCode(OperationType::Move));
+    CPPUNIT_ASSERT_EQUAL(std::string("Edit(4)"), toStringWithCode(OperationType::Edit));
+    CPPUNIT_ASSERT_EQUAL(std::string("Delete(8)"), toStringWithCode(OperationType::Delete));
+    CPPUNIT_ASSERT_EQUAL(std::string("Rights(16)"), toStringWithCode(OperationType::Rights));
+    CPPUNIT_ASSERT(Utility::startsWith(toStringWithCode(static_cast<NodeType>(0x09)), noConversionStr));
```
Comment (herve-er) [inline test/libcommon/utility/testtypes.cpp:48]:
```suggestion
    CPPUNIT_ASSERT(Utility::startsWith(toStringWithCode(fromInt<OperationType>(0x09)), noConversionStr));
```

File: test/libcommon/utility/testtypes.cpp [hunk 5]
```diff
@@ -34,6 +36,17 @@ void TestTypes::testStreamConversion() {
     CPPUNIT_ASSERT_EQUAL(std::string("Unknown(0)"), toStringWithCode(NodeType::Unknown));
     CPPUNIT_ASSERT_EQUAL(std::string("File(1)"), toStringWithCode(NodeType::File));
     CPPUNIT_ASSERT_EQUAL(std::string("Directory(2)"), toStringWithCode(NodeType::Directory));
+    CPPUNIT_ASSERT_EQUAL(std::string("Directory(2)"), toStringWithCode(static_cast<NodeType>(2)));
+    CPPUNIT_ASSERT(Utility::startsWith(toStringWithCode(static_cast<NodeType>(3)), noConversionStr));
```
Comment (herve-er) [inline test/libcommon/utility/testtypes.cpp:40]:
```suggestion
    CPPUNIT_ASSERT_EQUAL(std::string("Directory(2)"), toStringWithCode(fromInt<NodeType>(2)));
    CPPUNIT_ASSERT(Utility::startsWith(toStringWithCode(fromInt<NodeType>(3)), noConversionStr));
```

File: test/libcommon/utility/testtypes.cpp (hunk 4 above)
Comment (ClementKunz) [reply test/libcommon/utility/testtypes.cpp]: Done

File: test/libcommon/utility/testtypes.cpp (hunk 5 above)
Comment (ClementKunz) [reply test/libcommon/utility/testtypes.cpp]: Done

File: test/libcommon/utility/testtypes.cpp [hunk 6]
```diff
@@ -34,6 +36,17 @@ void TestTypes::testStreamConversion() {
...
+    CPPUNIT_ASSERT_EQUAL(std::string("Directory(2)"), toStringWithCode(fromInt<NodeType>(2)));
+    CPPUNIT_ASSERT(Utility::startsWith(toStringWithCode(fromInt<NodeType>(3)), noConversionStr));
+
+    CPPUNIT_ASSERT_EQUAL(std::string("None(0)"), toStringWithCode(OperationType::None));
+    CPPUNIT_ASSERT_EQUAL(std::string("Create(1)"), toStringWithCode(OperationType::Create));
+    CPPUNIT_ASSERT_EQUAL(std::string("Move(2)"), toStringWithCode(OperationType::Move));
+    CPPUNIT_ASSERT_EQUAL(std::string("Edit(4)"), toStringWithCode(OperationType::Edit));
+    CPPUNIT_ASSERT_EQUAL(std::string("Delete(8)"), toStringWithCode(OperationType::Delete));
+    CPPUNIT_ASSERT_EQUAL(std::string("Rights(16)"), toStringWithCode(OperationType::Rights));
+    CPPUNIT_ASSERT(Utility::startsWith(toStringWithCode(fromInt<NodeType>(0x09)), noConversionStr));
```
Comment (herve-er) [inline test/libcommon/utility/testtypes.cpp:48]:
```suggestion
    CPPUNIT_ASSERT(Utility::startsWith(toStringWithCode(fromInt<OperationType>(0x09)), noConversionStr));
```

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Review by ChristopheLarchier — APPROVED** [review]

**Pull request merged**